rust/azure_iot_operations_services/tests/azure_device_registry_client_tests.rs

```diff
@@ -0,0 +1,970 @@
+            let mut observation = azure_device_registry_client
+                .observe_device_update_notifications(
+                    DEVICE1.to_string(),
+                    ENDPOINT1.to_string(),
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Device update observation: {observation:?}");
+
+            let first_notification_notify = Arc::new(Notify::new());
+            let cleanup_notify = Arc::new(Notify::new());
+            let receive_notifications_task = tokio::task::spawn({
+                let first_notification_notify = first_notification_notify.clone();
+                let cleanup_notify = cleanup_notify.clone();
+
+                async move {
+                    log::info!("[{log_identifier}] Device update notification receiver started.");
+                    let mut count = 0;
+                    let mut first_notification_sent = false;
+                    loop {
+                        tokio::select! {
+                            notification_result = observation.recv_notification() => {
+                                if let Some((device, _)) = notification_result {
+                                    count += 1;
+                                    log::info!("[{log_identifier}] Device Observation received: {device:?}");
+
+                                    if count == 1 {
+                                        // Signal that we got the first notification
+                                        if !first_notification_sent {
+                                            first_notification_notify.notify_one();
+                                            first_notification_sent = true;
+                                        }
+                                        assert_eq!(device.name, DEVICE1);
+                                    } else {
+                                        log::info!("[{log_identifier}] Device Observation unexpected: {device:?}");
+                                        // Should not receive more than 1 notification
+                                        assert!(count < 2, "Received unexpected additional device observation notification");
+                                    }
+                                } else {
+                                    // Channel closed
+                                    break;
+                                }
+                            }
+                            () = cleanup_notify.notified() => {
+                                // We're told to stop waiting for notifications
+                                log::info!("[{log_identifier}] Cleanup signal received, stopping notification receiver");
+                                break;
+                            }
+                        }
+                    }
+
+                    // After loop exits for ANY reason:
+                    if !first_notification_sent {
```

Comment:
what is this scenario for? Since you have the timeout on not receiving the `first_notification_notify`, I don't think you'd need this safeguard?